plone/app/uuid/utils.py

```diff
@@ -1,18 +1,33 @@
 # -*- coding: utf-8 -*-
 from Products.CMFCore.utils import getToolByName
+from Products.ZCatalog.query import IndexQuery
 from zope.component.hooks import getSite
 
 
 def uuidToPhysicalPath(uuid):
     """Given a UUID, attempt to return the absolute path of the underlying
     object. Will return None if the UUID can't be found.
-    """
 
-    brain = uuidToCatalogBrain(uuid)
-    if brain is None:
+    This version is four times faster than the original.
+    """
+    site = getSite()
+    if site is None:
         return None
-
-    return brain.getPath()
+    catalog = getToolByName(site, 'portal_catalog', None)
+    if catalog is None:
+        return None
+    index = catalog.Indexes["UID"]
+    # This works even faster, but uses a private attribute:
+    # rid = index._index.get(uuid)
```

Comment:
IMO it is save to use `_index`.

Reply:
I try `_index` now, and fall back to `IndexQuery` when it fails.